src/osd/scheduler/mClockScheduler.cc

```diff
@@ -103,70 +161,68 @@ const dmc::ClientInfo *mClockScheduler::ClientRegistry::get_info(
   }
 }
 
-void mClockScheduler::set_max_osd_capacity()
+void mClockScheduler::set_max_osd_random_write_iops()
 {
   if (is_rotational) {
-    max_osd_capacity =
+    max_osd_random_write_iops =
       cct->_conf.get_val<double>("osd_mclock_max_capacity_iops_hdd");
     cct->_conf.set_val("osd_mclock_max_capacity_iops_ssd", "0");
   } else {
-    max_osd_capacity =
+    max_osd_random_write_iops =
       cct->_conf.get_val<double>("osd_mclock_max_capacity_iops_ssd");
     cct->_conf.set_val("osd_mclock_max_capacity_iops_hdd", "0");
   }
+  max_osd_random_write_iops = std::max<double>(
+    1.0, // ensure iops is non-zero and positive
```

Comment:
IIRC the config subsystem had / still has the _validator_ concept. Maybe it would be better to inform a user about a miconfiguration instead silently ignoring it?

(not a blocker)